### PR TITLE
Cleanup TLS setup document

### LINKED ISSUE #4959: Document how to use minio with TLS using IP address
Even after following the TLS setup guide, I can't get TLS to work with a local minio server and hitting `127.0.0.1`.  

This is the error I get:
```
ERRO[0006] Error in reading from new TLS connection 127.0.0.1:61985 at server 127.0.0.1:9001  cause=EOF source=[listener.go:188:github.com/minio/minio/pkg/http.(*httpListener).start.func2()]
```

I would rather not run DNS for this local app.  Is there a way to configure TLS using a static IP address?

## Your Environment

* Version used (`minio version`):
```
Version: 2017-08-05T00:00:53Z
Release-Tag: RELEASE.2017-08-05T00-00-53Z
Commit-ID: aeafe668d8b6d25caac671d59e2b0f0473ce35d0
```
* Environment name and version (e.g. nginx 1.9.1): running locally on a mac
* Server type and version: localhost
* Operating System and version (`uname -a`): 
`Darwin Ryans-MacBook-Pro.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64`
* Link to your project:

### PR DESCRIPTION
**Pull request by nitisht**
## Description
This updates the TLS documentation to be more precise. Also added steps on how to generate self signed certificates for an IP address using OpenSSL.

## Motivation and Context
Fixes  #4959

## How Has This Been Tested?
Manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/tls/README.md [hunk 1]
```diff
@@ -117,13 +139,14 @@ encryption_key
...
 ```
 
 ## 4. Install third-party CAs
 
 Minio can be configured to connect to other servers, whether Minio nodes or servers like NATs, Redis. If these servers use certificates that are not registered in one of the known certificates authorities, you can make Minio server trust these CAs by dropping these certificates under Minio config path (`~/.minio/certs/CAs/` on Linux or `C:\Users\<Username>\.minio\certs\CAs` on Windows).
 
 # Explore Further
-* [Minio Quickstart Guide](https://docs.minio.io/docs/minio-quickstart-guide)
 * [Minio Client Complete Guide](https://docs.minio.io/docs/minio-client-complete-guide)
+* [Generate Let's Encrypt Certificate](https://docs.minio.io/docs/generate-let-s-encypt-certificate-using-concert-for-minio)
```
Comment (harshavardhana) [inline docs/tls/README.md:151]: concert is deprecated we should remove, we should point to certbot.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: docs/tls/README.md (hunk 1 above)
Comment (nitisht) [reply docs/tls/README.md:151]: The link actually points to certbot doc, not sure why link says concert

Comment (harshavardhana) [reply docs/tls/README.md:151]: perhaps we should fix the slug name.

File: docs/tls/README.md [hunk 2]
```diff
@@ -1,38 +1,37 @@
...
 ## 1. Prerequisites
 
-* Download Minio server from [here](https://docs.minio.io/docs/minio-quickstart-guide)
+Download Minio server from [here](https://docs.minio.io/docs/minio-quickstart-guide)
 
-## 2. Configure with existing certificates
+## 2. Existing certificates
 
-Assuming that you are already having private and public certificates, you will need to copy them under `certs` in your Minio config directory using the names `private.key` and `public.crt` for key and public certificates respectively.
+If you have already acquired private and public certificates, copy them under `certs` directory in your Minio config directory (by default this is `/home/username/.minio/`). Note that the file names should be `private.key` and `public.crt` for key and public certificate respectively.
```
Comment (harshavardhana) [inline docs/tls/README.md]: We can say about windows home path as well? like we can say 

- `${HOME}/.minio/` 
- `%%USERPROFILE%%\.minio\` 

?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: docs/tls/README.md (hunk 2 above)
Comment (nitisht) [reply docs/tls/README.md]: sure, will do

Comment (nitisht) [reply docs/tls/README.md]: this is done

**Review by harshavardhana — DISMISSED** [review]
LGTM tested the new docs.

File: docs/tls/README.md [hunk 3]
```diff
@@ -46,11 +45,38 @@ Generate the self-signed certificate:
...
+
+[alt_names]
+IP.1 = 127.0.0.1
+```
+
+Then run
 
-Minio only supports key/certificate in PEM format on Windows. Currently we do not yet support PFX certificates.
+```sh
+openssl req -x509 -nodes -days 730 -newkey rsa:2048 -keyout private.key -out public.crt -config openssl.cnf
```
Comment (vadmeste) [inline docs/tls/README.md]: s/openssl.cnf/openssl.conf/

Comment (nitisht) [reply docs/tls/README.md]: updated @vadmeste

**Review by harshavardhana — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**